### PR TITLE
feat(GUI): add app to gui folder structure

### PR DESCRIPTION
**Pull request by Shou**
We add a `lib/gui/app/` folder to help transition to Webpack usage.

Change-Type: patch
Changelog-Entry: Add `lib/gui/app` folder to ease into Webpack usage.

### CODE REVIEW

File: tests/gui/pages/main.spec.js [hunk 1]
```diff
@@ -26,9 +27,17 @@ const availableDrives = require('../../../lib/shared/models/available-drives')
 const selectionState = require('../../../lib/shared/models/selection-state')
 require('angular-mocks')
 
+// Mock HTML requires by reading from the file-system
+// eslint-disable-next-line node/no-deprecated-api
+require.extensions['.html'] = (module, filename) => {
```
Comment (jviotti) [inline tests/gui/pages/main.spec.js]: Is this needed for this particular PR? (I think this is more of a Webpack change)

Comment (Shou) [reply tests/gui/pages/main.spec.js]: I swear I removed that from the patch file... good catch though!

**Comment by Shou** [comment]
@jviotti done

**Comment by Shou** [comment]
What's going on with the tests here?

**Comment by jviotti** [comment]
@Shou The Concourse infrastructure is having serious connection issues when fetching npm dependencies :/

**Comment by jviotti** [comment]
@Shou We're getting some conflicts here :)

**Comment by Shou** [comment]
@jviotti here we go

**Pull request merged**